ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,488 @@
+
+
+RAY_SUBMISSION_ID = "_RAY_SUBMISSION_ID"
+
+config._add(
+    "ray",
+    dict(
+        _default_service=schematize_service_name("ray"),
+        resubmit_interval=float(_get_config("DD_TRACE_RAY_RESUBMIT_LONG_RUNNING_INTERVAL", default=120.0)),
+        watch_delay=float(_get_config("DD_TRACE_RAY_WATCH_LONG_RUNNING_DELAY", default=10.0)),
+    ),
+)
+
+
+def _supported_versions() -> Dict[str, str]:
+    return {"ray": ">=2.46.0"}
+
+
+def get_version() -> str:
+    return str(getattr(ray, "__version__", ""))
+
+
+class RayTraceProcessor:
+    def process_trace(self, trace: List[Span]) -> Optional[List[Span]]:
+        if not trace:
+            return trace
+
+        filtered_spans = []
+        for span in trace:
+            if span.service == "ray.dashboard" and span.get_tag("component") != "ray":
+                continue
+
+            if span.get_tag("component") == "ray":
+                span.set_metric(_DJM_ENABLED_KEY, 1)
+                span.set_metric(_FILTER_KEPT_KEY, 1)
+                span.set_metric(_SPAN_MEASURED_KEY, 1)
+                span.set_metric(_SAMPLING_PRIORITY_KEY, 2)
+
+                # add host name for GPU Monitoring correlation
+                span.set_tag_str(_HOSTNAME_KEY, socket.gethostname())
+
+            filtered_spans.append(span)
+        return filtered_spans
+
+
+def _wrap_remote_function_execution(function):
+    """Inject trace context parameter into function signature"""
+
+    @wraps(function)
+    def wrapped_function(*args, **kwargs):
+        return _wrap_task_execution(function, *args, **kwargs)
+
+    return wrapped_function
+
+
+def _wrap_task_execution(wrapped, *args, **kwargs):
+    """
+    Wraps the actual execution of a Ray task to trace its performance.
+    """
+    if not tracer or "_dd_trace_ctx" not in kwargs:
+        return wrapped(*args, **kwargs)
+
+    # Extract context from parent span
+    extracted_context = _TraceContext._extract(kwargs["_dd_trace_ctx"])
+    kwargs.pop("_dd_trace_ctx")
+
+    function_name = getattr(wrapped, "__name__", "unknown_function")
+    function_module = getattr(wrapped, "__module__", "unknown_module")
+
+    with long_running_ray_span(
+        f"{function_module}.{function_name}",
+        service=os.environ.get(RAY_SUBMISSION_ID),
+        span_type=SpanTypes.RAY,
+        child_of=extracted_context,
+        activate=True,
+    ) as task_execute_span:
+        try:
+            set_maybe_big_tag(task_execute_span, "ray.task.args", args)
+            set_maybe_big_tag(task_execute_span, "ray.task.kwargs", kwargs)
+
+            result = wrapped(*args, **kwargs)
+
+            task_execute_span.set_tag_str("ray.task.status", "success")
+            return result
+        except Exception:
+            task_execute_span.set_tag_str("ray.task.status", "error")
+            raise
+
+
+def traced_submit_task(wrapped, instance, args, kwargs):
+    """Trace task submission, i.e the func.remote() call"""
+
+    if not tracer:
```

Comment:
Do we need this? I don't think tracer is ever None

Reply:
I was thinking it was the standard in the integrations. It appears it is not, so I remove it everywhere